Patient Registry System/PatientRegistrySystem.Services/UserService.cs

```diff
@@ -0,0 +1,71 @@
+﻿using AutoMapper;
+using Microsoft.AspNetCore.Mvc;
+using PatientRegistrySystem.DB.Contexts;
+using PatientRegistrySystem.DB.Entities;
+using PatientRegistrySystem.DB.Repos;
+using PatientRegistrySystem.Domain.Dto;
+using System;
+using System.Linq;
+
+namespace PatientRegistrySystem.Services
+{
+    public class UserService : Controller, IUserService
+    {
+        public IMapper Mapper { get; }
+        public IGenericRepository<User> userRepository { get; }
+
+        public UserService(IMapper mapper, IGenericRepository<User> UserGenericRepository)
+        {
+            Mapper = mapper ?? throw new ArgumentNullException(nameof(mapper));
+            userRepository = UserGenericRepository ?? throw new ArgumentNullException(nameof(UserGenericRepository));
+        }
+        //CreateUser/UpdateUser/DeleteUser/GetUser/GetAllUsers
+
+        public IActionResult GetAllUsers()
+        {
+            var usersFromRepo = userRepository.GetAll();
```

Comment:
can you please double check the repository methods you have. they should use the `Async` version of queries.
https://docs.microsoft.com/en-us/ef/core/miscellaneous/async